File: AffiliateMarketing.Domain/Entities/AffiliateProfile.cs
namespace AffiliateMarketing.Domain.Entities
{
    public sealed class AffiliateProfile
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public required string TrackingId { get; set; }
    public DateTime CreatedAt { get; set; }
}
}

File: affiliate_api/app/models/affiliate_profile.py
# Source: AffiliateMarketing.Domain/Entities/AffiliateProfile.cs
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import mapped_column, Mapped
from app.core.db import Base

class AffiliateProfile(Base):
    __tablename__ = "affiliate_profiles"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    user_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
    tracking_id: Mapped[str] = mapped_column(String, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
